src/figuresClasses.ts

```diff
@@ -1,19 +1,120 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+abstract class BaseFigure implements Figure {
+  abstract readonly shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
+
+  protected checkAllSides(sides: number[]): void {
+    if (sides.some((side) => side <= 0)) {
+      throw new Error(
+        `Figure of shape ${this.shape} should have only positive params`,
+      );
+    }
+  }
+
+  static roundNumber(value: number): number {
+    return Number(value.toFixed(2));
+  }
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    this.checkAllSides([a, b, c]);
 
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = [a, b, c].sort((prev, next) => prev - next);
+
+    if (sideC >= sideA + sideB) {
+      throw new Error('Wrong values of sides');
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c),
+    );
+
+    return BaseFigure.roundNumber(area);
+  }
 }
 
-export class Rectangle {
+export class Circle extends BaseFigure {
+  readonly shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    super();
+    this.checkAllSides([radius]);
 
+    if (radius <= 0) {
+      throw new Error('Wrong values of sides');
+    }
```

Comment:
The duplicate operation, you have already checked the radius with `this.checkAllSided`.